microhapulator/api.py

```diff
@@ -578,61 +579,89 @@ def type(bamfile, markertsv, minbasequal=10, max_depth=1e6):
 
 
 def read_length_dist(
-    fastq,
+    lengthfiles,
     plotfile,
-    lengthsfile=None,
+    samples,
+    hspace=-0.7,
     xlabel="Read Length (bp)",
-    xlim=None,
-    scale=1000,
     title=None,
-    color=None,
-    edgecolor=None,
+    color="red",
+    edgecolor="black",
 ):
     """Plot distribution of read lengths
-
-    :param str fastq: path of a FASTQ file containing NGS reads
+    :param list lengthsfiles: list of JSON files containing read lengths for each sample
     :param str plotfile: path of a graphic file to create
-    :param str lengthsfile: if specified, read lengths will be written to the specified file
     :param str xlabel: label for the X axis
-    :param tuple xlim: a 2-tuple of numbers (x1, x2) representing the start and end points of the portion of the X axis to be displayed; by default this is determined automatically
-    :param float scale: scaling factor for the Y axis
     :param str title: title for the plot
     :param str color: override histogram plot color; red by default
-    :param str edgecolor: override histogram edge color; dark red by default
+    :param str edgecolor: override histogram edge color; black by default
     """
+    read_lengths = aggregate_read_lengths(lengthfiles, samples)
     backend = matplotlib.get_backend()
     plt.switch_backend("Agg")
+    sns.set_theme(style="white", rc={"axes.facecolor": (0, 0, 0, 0), "axes.linewidth": 2})
+    grid = sns.FacetGrid(read_lengths, row="Sample", hue="Sample", aspect=20)
+    grid.map_dataframe(sns.kdeplot, x="ReadLength", color=color, fill=True, alpha=1)
+    grid.map_dataframe(sns.kdeplot, x="ReadLength", color=edgecolor)
+    grid.map(read_distribution_plot_label, "Sample")
```

Comment:
This `grid.map()` function here is necesarry to label each subplot with the sample name.  See my comment on the  `read_distribution_plot_label()` function below.